spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/support/DatastoreRepositoryFactory.java

```diff
@@ -82,4 +103,39 @@ protected Object getTargetRepository(RepositoryInformation metadata) {
 	protected Class<?> getRepositoryBaseClass(RepositoryMetadata metadata) {
 		return SimpleDatastoreRepository.class;
 	}
+
+	@Override
+	protected Optional<QueryLookupStrategy> getQueryLookupStrategy(@Nullable Key key,
+			QueryMethodEvaluationContextProvider evaluationContextProvider) {
+
+		return Optional.of(new DatastoreQueryLookupStrategy(this.datastoreMappingContext,
+				this.datastoreTemplate,
+				delegateContextProvider(evaluationContextProvider), EXPRESSION_PARSER));
+	}
+
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext)
+			throws BeansException {
+		this.applicationContext = applicationContext;
+	}
+
+	private QueryMethodEvaluationContextProvider delegateContextProvider(
```

Comment:
This seems only need for SpEL.

Reply:
My reasoning for keeping this one is:

1. This private method is needed by `getQueryLookupStrategy`, which is a function we must override to use query methods (the parent version returns nothing). 
2. In that function we are given the `QueryMethodEvaluationContextProvider` (which i infer is because the Spring Data folks expected SpEL to be used in query methods), so we might as well keep it and pass it down even if we aren't going to use it right now.
3. then this private function just adds this object's instance of the applicationContext to that  `QueryMethodEvaluationContextProvider`. Nothing specifically SpEL about doing this, and if we are going to not throw away that `QueryMethodEvaluationContextProvider` then we might as well also add this.applicationContext to it.